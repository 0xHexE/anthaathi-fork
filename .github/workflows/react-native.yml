name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
