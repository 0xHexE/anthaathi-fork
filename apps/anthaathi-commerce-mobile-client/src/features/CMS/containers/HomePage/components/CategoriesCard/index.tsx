import {Image, View} from 'react-native';
import React from 'react';
import {Text, TouchableRipple} from 'react-native-paper';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';
import {HomePageComponentType} from '../../../../types/common';

export interface CategoryProps {
  key: string;
  title: string;
  image: string;
}

export interface CategoriesCardProps {
  title?: string;
  categories: CategoryProps[];
}

const CategoriesCard = (props: CategoriesCardProps) => {
  return (
    <View style={{marginHorizontal: 10}} testID="categoryCard">
      <Text variant="titleLarge" style={{marginBottom: 9, fontSize: 20}}>
        {props.title}
      </Text>

      <View
        style={{
          display: 'flex',
          flexDirection: 'row',
          flexWrap: 'wrap',
        }}>
        {props.categories.map(item => {
          return <CategoryItemRenderer key={item.key} item={item} />;
        })}
      </View>
    </View>
  );
};

function CategoryItemRenderer({item}: {item: CategoryProps}) {
  const itemHeight = useResponsiveValue([180, 240, 360, 360]);
  const itemWidth = useResponsiveValue(['98%', '48%', '31%', '23%']);

  return (
    <TouchableRipple onPress={() => {}}>
      <View
        style={{
          height: itemHeight,
          width: itemWidth,
          marginVertical: 10,
          marginHorizontal: '1%',
        }}
        key={item.key}>
        <View>
          <Image
            testID="categoryImage"
            source={{
              uri: item.image,
            }}
            style={{height: '85%', width: '100%'}}
          />
          <Text
            style={{
              textAlign: 'center',
              marginBottom: 12,
              marginTop: 6,
              fontWeight: '600',
            }}
            variant="labelMedium">
            {item.title}
          </Text>
        </View>
      </View>
    </TouchableRipple>
  );
}

export default CategoriesCard;

export const CategoriesCardCMSInput = {
  _component: HomePageComponentType.CategoriesCard,
  component: CategoriesCard,
};
