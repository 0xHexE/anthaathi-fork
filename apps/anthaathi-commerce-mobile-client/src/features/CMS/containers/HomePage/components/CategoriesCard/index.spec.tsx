import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import CategoriesCard from './index';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';

describe('CategoryCard', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <CategoriesCard
            title="Categories"
            categories={[
              {
                title: 'test',
                key: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('categoryCard')).toBeTruthy();
  });

  it('should have CategoryCard title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <CategoriesCard
            title="Categories"
            categories={[
              {
                title: 'test',
                key: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );
    expect(temp.queryByText('Categories')).toBeTruthy();
  });

  it('should have Categories Image', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <CategoriesCard
            title="Categories"
            categories={[
              {
                title: 'test',
                key: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('categoryImage')!.props.source).toMatchObject({
      uri: 'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
    });
  });
});
