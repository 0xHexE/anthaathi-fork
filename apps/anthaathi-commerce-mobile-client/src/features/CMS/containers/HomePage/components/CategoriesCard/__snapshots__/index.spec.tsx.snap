// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should render the item 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
  testID="categoryCard"
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {
          "fontSize": 20,
          "marginBottom": 9,
        },
      ]
    }
  >
    Categories
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "height": 240,
            "marginHorizontal": "1%",
            "marginVertical": 10,
            "width": "48%",
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
              }
            }
            style={
              Object {
                "height": "85%",
                "width": "100%",
              }
            }
            testID="categoryImage"
          />
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "letterSpacing": 0.5,
                  "lineHeight": 16,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "fontWeight": "600",
                  "marginBottom": 12,
                  "marginTop": 6,
                  "textAlign": "center",
                },
              ]
            }
          >
            test
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
